pkg/util/permissions/permissions_test.go

```diff
@@ -0,0 +1,74 @@
+package permissions
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/authorization/mgmt/2015-07-01/authorization"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+func TestCanDoAction(t *testing.T) {
+	tests := []struct {
+		name        string
+		permissions []authorization.Permission
+		action      string
+		want        bool
+		wantErr     bool
+	}{
+		{
+			name:        "empty permissions list",
+			permissions: []authorization.Permission{},
```

Comment:
you should be able to omit all items like `[]type{}`, so please do - in that case Go will use nil; note that len and range on nil slices and maps in Go is legal per the spec.